src/components/results/body/index.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { TREE_VIEW, JSON_VIEW } from '../results-view-selector';
+import JSONTree from 'react-json-tree';
+import { escapeLikeJSON, stringify } from '../utils';
+
+import './style.css';
+
+const expandFalse = () => false;
+
+const RequestBody = ( { response, view } ) => {
+
+	const getItemString = ( type, data, itemType, itemString ) => (
+			/* eslint-disable react/no-danger */
+		<span
+			className="collapsed-content"
+			dangerouslySetInnerHTML={ { __html: `${ itemString } <span class="content">${ stringify( data ) }</span>` } }
+		/>
+			/* eslint-enable react/no-danger */
+		);
+
+	const jsonTheme = {
+		extend: 'default',
+		tree: {
+			MozUserSelect: 'text',
+			WebkitUserSelect: 'text',
+		},
+		nestedNodeItemString: ( node, expandedNodes, type, expanded ) => {
+			return {
+				className: expanded ? 'expanded' : 'collapsed',
+			};
+		},
+	};
+
+	// It adds double quotes when rendering strings values.
+	const customStringRenderer = value => {
+		if ( typeof value === 'string' ) {
+			const valueWithoutQuotes = value.replace( /^"|"$/g, '' );
+			let valueEscaped;
```

Comment:
the way we're trying to infer types here is a bit confusing. `valueRenderer` gets multiple arguments though which can help:

```ts
const valueRenderer = (
	/** As it appears in the JSON, sometimes missing for some reason */
	valueAsString: string | null?,

	/** Parsed JSON value, with expected JS type */
	value: JSONValue,

	/** Path into original JSON value to get to here */
	...pathKey: string[]
) => { … }
```

That means we should be able to simplify this function and avoid all the awkward parsing.

```ts
const valueRenderer = ( valueAsString, value ) =>
	'string' === typeof value
		? `"${ escapeLikeJSON( value ) }"`
		: valueAsString;
```

If this is too much for this PR that's fine. Would at least be a nice follow-up.

Reply:
Yeah! It seems to work nicely. I didn't plan to refactor/improve that function as it was some kind of black box for me and not related to the change I meant to add, but I think it's a nice touch too.

I haven't seen any regression issue, but if you could double-check any other endpoint you might think it's conflictive it would be great, as we are not returning the same data as before because sometimes we were returning a span with some classes added to it.

Reply:
that's a good catch and the span is something I just completely overlooked. the full version with it isn't much more complicated though.

```ts
const valueRenderer = ( valueAsString, value ) => {
	// when and why does this happen?
	if ( 'string' !== typeof valueAsString ) {
		return valueAsString;
	}

	const displayValue = 'string' === typeof value
		? `"${ escapeLikeJson( value ) }"`
		: valueAsString;

	return <span className="expanded-value">{ displayValue }</span>
}
```